### DISCUSSION TITLE
Raid Stats

### DISCUSSION THREAD

**Discussion by nobrayner** [OpenSourceRaidGuild/tutorial-quest#2] [Ideas]
# Overview

To really get the gamification of these raids going, we will need to figure out some stats. There are two types of "Stats" that I want to address:
* Contribution Stats
_These are stats like Commits, Additions, Deletions, Reviews, etc._
* "Character" Stats
_These will be based off the D&D 5e Stats - Strength, Dexterity, Constitution, Wisdom, Intelligence, Charisma_

## Contribution Stats

The current thoughts on Contribution stats are as follows

### Raid Overview Stats

* Total Contributors
* Total Additions
* Total Deletions
* Total Commits
* Total Files Changed
* Raid duration (in days, or hours if less than a day)

### Per User stats:
* Number of Commits
  * Exclude merge commits
  * Exclude non-raid commits (upstream)
* Total Additions
* Total Deletions
* Total (PR) Reviews
  * **!! Need some good rules around this !!**

## "Character" Stats

The aim of the Character stats is to attribute them to a raider's "Character". That is to say, you can increase your characters stats by participating in raids. The increase amount, and which stat is increased, will be affected by the contribution stats from the raids participated in.

# What is needed from you

We need you guys to discuss and ideate on:
1. More Contributions Stats, and the rules around them; and then secondarily
2. How these Contribution Stats should feed into the RPG Stats

**Comment by tigerabrodi** [comment]
- User Stats? How do we get each user's information? Do they have to do something themselves or, i.e. sign up? 
- How we exactly calculate the "Character" stats based on the user's stats (his/her contributions), that will also be something we need to bring up later.
- I like it overall, as for "character" stats, I bet users would have to sign up.

**Reply by tigerabrodi** [reply]
Regards to my first point, I bet users would have to sign up, otherwise, they can not have a character. :+1: 

Seems good, very good, well done buddy, kudos! :fire: :raised_hands:

**Reply by nobrayner** [reply]
RE your second point: The "character" stats will have to be calculated based on an individual user's contribution stats - at the end of each raid, the contribution stats get converted to character stat increases. Since these "character" stats will be directly affected by the contribution stats, it is worth discussing now as well.

**Reply by tigerabrodi** [reply]
Sounds good, what @codyarose mentioned below is a great step :fire:

**Reply by nobrayner** [reply]
Addressing your point about signing up - currently stats would be tracked based on GitHub login, so theoretically you already signed up :wink:

Character stats would just be associated back to the persons GitHub account.

**Reply by tigerabrodi** [reply]
@nobrayner :tada: This makes the Sign Up much easier if through Github :fire:, awesome :raised_hands:

**Comment by codyarose** [comment]
Another per user stat: number of raids participated in.
I'm not super familiar with the D&D 5e stuff, so I'm interested in other people's input/interpretation on that. However, if I had to take a stab at it I would say:
- PR Reviews -> Charisma: seems like the obvious choice, but maybe also Wisdom?
- Deletions -> Intelligence: ability to recognize when something could be simplified or removed?
- Commits || Additions -> Strength || Dexterity: can't think of a metaphor, but it feels right
- if # of raids participated in is added || Commits -> Constitution: the stamina to keep going back into the fight

**Reply by codyarose** [reply]
Also, obviously the values of the contribution stats will have to be adjusted somehow when transferring to character stats. We'll have _way_ more additions/deletions than commits, so either: 
- the value of commits should be higher than additions (0.1 pts vs 1.0 pts), or 
- the total number of additions needed to get to the next corresponding stat level should be much higher than the number of commits needed to get to the next level of whichever stat commits corresponds to

**Reply by nobrayner** [reply]
Definitely agree on needing some kind of "weighting" between different stats. :100: shouldn't be mapped 1:1

**Reply by emma-r-slight** [reply]
I think these are all great ideas, but how do make this include people who might not have the skills to make valuable contributions to code, but still want to be involved?  So they may be part of discussions or something in Discord, can we somehow make participation a thing?

**Reply by emma-r-slight** [reply]
I'm trying to keep this in mind:
- To make contributing to OSS enjoyable, engaging, and with a low barrier-to-entry for those who have never done it before

**Reply by nobrayner** [reply]
@emma-r-slight definitely... the running idea is to utilize the gamification to make it more fun, and a great community around it - thus the lower barrier to entry. However... the gamification will not be for everyone. I would love to see some sort of "dual stream" system.

For example  you have the "casual" stream where it is all about contributing at all, and then the "hardcore" stream where you get fully into the gamification system. But then of course making it so you can easily transition between the two if need be.

Maybe mode is a better term than stream..

**Reply by emma-r-slight** [reply]
That I think is a great idea! It is something I've been thinking about for a while.  On the home page have two "adventures/quests....whatever we call them" to choose from.... novice/squire/Noob option or Knight/warrior/experienced.  Noob path directs them to "tasks' you need to learn to level up or something.... making git commits, issues. PR, code reviews perhaps.. we could even have some easier repos to tackle in this one.? 

A lot more work though....

**Reply by JacobMGEvans** [reply]
I also have ideas for code contributing systems that would make ANYONE able to participate with a really low bar of entry and still potentially earn achievements if they wanted.

**Comment by sudokoi** [comment]
Are we letting users to choose a `character` for themselves? Each character can have a fixed multiplier again each of the stats i.e. Strength, Dexterity, Constitution, Wisdom, Intelligence, Charisma.

Based on their contribution on any raid, the stats will increase and can also level up once they reach some threshold.

The list of character class can be fixed and the stats multipliers can be based on the character class.

**Reply by JacobMGEvans** [reply]
Oh I didn't even consider a possible leveling/EXP concept we can even tie that into prestige systems (resetting after maximum level) which gets you an achievement  

@nobrayner was achievements mentioned in another document?

**Reply by nobrayner** [reply]
Oh damn! I had briefly thought about XP, but not really dug into the idea yet... I'm liking the sounds of this :smile:

@JacobMGEvans currently no document for Achievements... I was wanting to get stats sorted out first - directly applicable to moving forward quickly - and then come back to flesh out things like achievements.

**Comment by nobrayner** [comment]
Closing this discussion - mainly to move it to the website repo :stuck_out_tongue:
